beat/models/seismic.py

```diff
@@ -725,35 +873,68 @@ def get_formula(
                         wc.event_idx, wmap._mapid))
                 sources = [self.sources[wc.event_idx]]
 
-            self.synthesizers[wmap._mapid] = theanof.SeisSynthesizer(
-                engine=self.engine,
-                sources=sources,
-                targets=wmap.targets,
-                event=self.events[wc.event_idx],
-                arrival_taper=wc.arrival_taper,
-                arrival_times=wmap._arrival_times,
-                wavename=wmap.name,
-                filterer=wc.filterer,
-                pre_stack_cut=self.config.pre_stack_cut,
-                station_corrections=self.config.station_corrections)
-
-            synths, _ = self.synthesizers[wmap._mapid](self.input_rvs)
+            if wmap.config.freqdomain_include:
 
-            residuals = wmap.shared_data_array - synths
-
-            logpts = multivariate_normal_chol(
-                wmap.datasets, wmap.weights, hyperparams, residuals,
-                hp_specific=hp_specific)
-
-            wlogpts.append(logpts)
+                self.ffttransform[wmap._mapid] = theanof.FftTransform(
```

Comment:
you would add a flag to the seismic synthesizer that it knows wheter to do the fft_transforms, Please put the fft transforms into that theanoOp instead of adding that extra one